packages/jazz-react/src/index.tsx

```diff
@@ -128,24 +128,18 @@ export function createJazzReactContext<Acc extends Account>({
         id: ID<V> | undefined,
         depth: D & DepthsIn<V> = [] as D & DepthsIn<V>,
     ): DeeplyLoaded<V, D> | undefined {
-        // for some reason (at least in React 18) - if we use state directly,
-        // some updates get swallowed/UI doesn't update
-        const [_, setUpdates] = useState<number>(0);
-        const state = useRef<DeeplyLoaded<V, D> | undefined>(undefined);
+        const [state, setState] = useState<{ value: DeeplyLoaded<V, D> | undefined }>({ value: undefined });
         const me = React.useContext(JazzContext)?.me;
 
         useEffect(() => {
             if (!id || !me) return;
-            return subscribeToCoValue(Schema, id, me, depth, (update) => {
-                state.current = update as DeeplyLoaded<V, D>;
-
-                setUpdates((u) => {
-                    return u + 1;
-                });
+        
+            return subscribeToCoValue(Schema, id, me, depth, (value) => {
+                setState({ value })
```

Comment:
Using an object to hold the value reference should fix the update issues as it will trigger a re-render at each update (pretty much similar to the previous force update hack)

The difference is that now the value is committed inside the state, so it should be more compliant with the react conccurent requirements.